src/lib.rs

```diff
@@ -50,25 +50,23 @@ impl Preprocessor for D2 {
                 cmark(events, &mut buf).unwrap();
                 chapter.content = buf;
             }
-        });
-
-        Ok(book)
+        })
```

Comment:
right now, your `Preprocessor::run` method isn't returning anything, but it needs to return a `Result<Book, Error>`.

try adding an `Ok(book)` statement at the end (which is itself shorthand for `Result::<Book, Error>::Ok(book)`)

Reply:
Cool, done that